### DISCUSSION TITLE
Create an instace of a detailed grid to be used in another master dialog

### DISCUSSION THREAD

**Discussion by flashyjunior** [serenity-is/Serenity#7305] [Help] [answered]
Hi,
How can I create an instance of a detail editor `xyzeditor.tsx` already linked to  a master dialog with Columns A,B,C,D.
I want to use the same `xyzeditor.tsx` in another master dialog but with fewer columns (A and C) than the first master dialog.
And also want to hide the `Add Item` button in the new instance

Master Dialog

![image](https://github.com/user-attachments/assets/ad13bfa4-8cdd-44b0-8885-a80f8e7e6cc3)

New Master Dialog (Same Detail grid)

![image](https://github.com/user-attachments/assets/1ae6f04f-eb47-4734-9696-eb8f94f441c3)

**Comment by flashyjunior** [comment]
@minhhungit  , please can you assist me.
Thank you

**Reply by arunkumar12712** [reply, marked as answer]
create another tsx file 

import { Decorators } from "@serenity-is/corelib";
@Decorators.registerEditor('Test2Editor')
export class Test2Editor extends Test1Editor {


    protected getColumns() {
        var columns = super.getColumns();
        //Hide fields
       columns = columns.filter(x => x.field != "Some field")
        return columns;
    }
}

**Reply by flashyjunior** [reply]
Thank you, it worked
